### PR TITLE
Add submodule and subtree content

### PR DESCRIPTION
**Pull request by beardofedu**
This pull request will add submodule and subtree content to the training manual based on #240

### CODE REVIEW

File: docs/_sidebar.md [hunk 1]
```diff
@@ -40,6 +40,8 @@
 - Advanced Topics
   - [Personal Access Tokens](20_personal_access_token.md)
   - [.gitignore File](23_gitignore.md)
+  - [Git submodules](26_git_submodules.md)
+  - [Git subtrees](27_git_subtrees.md)
```
Comment (sujeetkb2014) [inline docs/_sidebar.md:45]: Added advanced topics

**Review by sujeetkb2014 — DISMISSED** [review]

**Review by allthedoll — DISMISSED** [review]
🌈

**Comment by allthedoll** [comment]
@beardofedu we need to add subtree knowledge, want to pair or do you have an ETA?

**Comment by beardofedu** [comment]
@allthedoll , yea, we can pair on this.

**Comment by 007Salih2020** [comment]
done

File: docs/27_git_subtrees.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+Subtrees are a git tool for including the content of another repository as a dependency of your project. As an example, if you are working on a repository that would benefit from having the contents of another repository, subtrees might be the tool you need. Unlike submodules which require the use of a `.gitmodule` file to manage the inclusion of another repository in your repository, subtrees don't rely on external files to be managed. 
+
+
+### Activity: Adding a subtree
+
+#### Setting up the empty repository for a subtree merge
+
+1. Open your preferred command line tool.
+2. Create a new directory and navigate to it.
+
```
Comment (007Salih2020) [inline docs/27_git_subtrees.md:12]: done

File: docs/26_git_submodules.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+## Git submodules
+
+In some instances, you might want to store the contents of another repository within your repository based on a specific commit in that other repository's history. Git provides a functionality to cover this need in the form of Git submodules.
+
+Git submodules enable you to include the contents of another repository based on a specific commit. This means while changes occur in the other repository you are able to continue utilize that project's history at a specific point.
```
Comment (007Salih2020) [inline docs/26_git_submodules.md:5]: done

**Review by 007Salih2020 — COMMENTED** [review]
done

**Comment by 007Salih2020** [comment]
done

**Review by amadarsh — APPROVED** [review]
Changes looks good

**Review by amadarsh — DISMISSED** [review]

File: docs/27_git_subtrees.md [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+Although we've only added one subproject, any number of subprojects can be incorporated into a Git repository.
+
+?> **Tip**: If you create a fresh clone of the repository in the future, the remotes you've added will not be created for you. You will have to add them again using [the `git remote add` command](https://docs.github.com/github/using-git/adding-a-remote).
+
+#### Synchronizing with updates and changes
+
+When a subproject is added, it is not automatically kept in sync with the upstream changes. You will need to update the subproject with the following command:
+
+```sh
+git pull -s subtree <em>remotename</em> <em>branchname</em>
```
Comment (parkerbxyz) [inline docs/27_git_subtrees.md:90]:
```suggestion
git pull -s subtree remotename branchname
```

**Review by crbonato — APPROVED** [review]

File: docs/26_git_submodules.md (hunk 3 above)
Comment (parkerbxyz) [inline docs/26_git_submodules.md:5]:
```suggestion
Git submodules enable you to include the contents of another repository based on a specific commit. This means while changes occur in the other repository you are able to continue to utilize that project's history at a specific point.
```

File: docs/28_git_subtrees.md [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+Although we've only added one subproject, any number of subprojects can be incorporated into a Git repository.
+
+?> **Tip**: If you create a fresh clone of the repository in the future, the remotes you've added will not be created for you. You will have to add them again using [the `git remote add` command](https://docs.github.com/github/using-git/adding-a-remote).
+
+#### Synchronizing with updates and changes
+
+When a subproject is added, it is not automatically kept in sync with the upstream changes. You will need to update the subproject with the following command:
+
+```sh
+git pull -s subtree <em>remotename</em> <em>branchname</em>
```
Comment (parkerbxyz) [inline docs/28_git_subtrees.md:90]:
```suggestion
git pull -s subtree <REMOTE_NAME> <BRANCH_NAME>
```

File: docs/27_git_submodules.md [hunk 6]
```diff
@@ -0,0 +1,5 @@
+## Git submodules
+
+In some instances, you might want to store the contents of another repository within your repository based on a specific commit in that other repository's history. Git provides a functionality to cover this need in the form of Git submodules.
+
+Git submodules enable you to include the contents of another repository based on a specific commit. This means while changes occur in the other repository you are able to continue utilize that project's history at a specific point.
```
Comment (parkerbxyz) [inline docs/27_git_submodules.md:5]:
```suggestion
Git submodules enable you to include the contents of another repository based on a specific commit. This means while changes occur in the other repository you are able to continue to utilize that project's history at a specific point.
```

**Review by parkerbxyz — APPROVED** [review]

**Review by tleumukan — APPROVED** [review]

**Pull request merged**